wrapper/src/test/java/integration/refactored/container/tests/HikariTests.java

```diff
@@ -115,4 +142,149 @@ public void testOpenConnectionWithDataSourceClassName() throws SQLException {
+
+      // Try to get a new connection to the failed instance, which times out
+      assertThrows(SQLTransientConnectionException.class, () -> dataSource.getConnection());
+    }
+  }
+
+  private HikariDataSource createDataSource(final Properties customProps) {
+    final HikariConfig config = getConfig(customProps);
+    HikariDataSource dataSource = new HikariDataSource(config);
+
+    final HikariPoolMXBean hikariPoolMXBean = dataSource.getHikariPoolMXBean();
+
+    LOGGER.fine("Starting idle connections: " + hikariPoolMXBean.getIdleConnections());
+    LOGGER.fine("Starting active connections: " + hikariPoolMXBean.getActiveConnections());
+    LOGGER.fine("Starting total connections: " + hikariPoolMXBean.getTotalConnections());
+    return dataSource;
+  }
+
+  private HikariConfig getConfig(Properties customProps) {
+    final HikariConfig config = new HikariConfig();
+    config.setUsername(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getUsername());
+    config.setPassword(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getPassword());
+    config.setMaximumPoolSize(3);
+    config.setReadOnly(true);
+    config.setExceptionOverrideClassName(HikariCPSQLException.class.getName());
+    config.setInitializationFailTimeout(75000);
+    config.setConnectionTimeout(1000);
+
+    config.setDataSourceClassName(AwsWrapperDataSource.class.getName());
+    config.addDataSourceProperty("targetDataSourceClassName", DriverHelper.getDataSourceClassname());
+    config.addDataSourceProperty("jdbcProtocol", DriverHelper.getDriverProtocol());
+    config.addDataSourceProperty("portPropertyName", "portNumber");
+    config.addDataSourceProperty("serverPropertyName", "serverName");
+    config.addDataSourceProperty("databasePropertyName", "databaseName");
+
+    Properties targetDataSourceProps = new Properties();
+
+    targetDataSourceProps.setProperty(
+        "serverName",
+        TestEnvironment.getCurrent()
+            .getInfo()
+            .getProxyDatabaseInfo()
+            .getInstances()
+            .get(0)
+            .getEndpoint());
+    targetDataSourceProps.setProperty(
+        "databaseName",
+        TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getDefaultDbName());
+
+    targetDataSourceProps.setProperty("portNumber",
+        Integer.toString(TestEnvironment.getCurrent().getInfo().getProxyDatabaseInfo().getClusterEndpointPort()));
+    DriverHelper.setSocketTimeout(targetDataSourceProps, SOCKET_TIMEOUT_SEC, TimeUnit.SECONDS);
+    DriverHelper.setConnectTimeout(targetDataSourceProps, CONNECT_TIMEOUT_SEC, TimeUnit.SECONDS);
+    DriverHelper.setMonitoringSocketTimeout(targetDataSourceProps, MONITORING_SOCKET_TIMEOUT_SEC, TimeUnit.SECONDS);
+    DriverHelper.setMonitoringConnectTimeout(targetDataSourceProps, MONITORING_CONNECT_TIMEOUT_SEC, TimeUnit.SECONDS);
+    targetDataSourceProps.setProperty(PropertyDefinition.PLUGINS.name, "failover,efm");
```

Comment:
These plugins should be used by the driver by default, so you can remove this line.